### PR TITLE
Integration of visibility into advanced mode mitaka

### PR DESCRIPTION
**Pull request by DhuldevValekar**
@njagadish @rajendramachani please review the changes.

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -26,6 +26,9 @@ Install stable/mitaka devstack with GBP.
        ConfiguratorQcow2Image=<configurator image path>
        VyosQcow2Image=<vyos image path>
        HaproxyQcow2Image=<haproxy image path>
+       
+       # if you want to Visibility then give image urls. 
+       VisibilityQcow2Image=
```
Comment (njagadish) [inline README.txt:31]: This is not visibility image. This has both configurator and visibility dockers. For now, we can keep this. Lets discuss and change in next version.

File: exercises/visibility_user_data/visibility-user-data [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#cloud-config
```
Comment (njagadish) [inline exercises/visibility_user_data/visibility-user-data]: The directory name should be user-data and the file name should be visibility_user_data. We need to change this.

File: lib/nfp [hunk 3]
```diff
@@ -145,66 +146,156 @@ function create_nfp_gbp_resources {
...
+sudo sed -i "s/rsyslog_host=*.*/rsyslog_host=$visibility_vm_ip/g" visibility-user-data
+cd $CUR_DIR
+
+}
 function upload_images_and_launch_configuratorVM {
     
     TOP_DIR=$1
     source $TOP_DIR/openrc neutron service
-    ConfiguratorQcow2ImageName=$2
-    VyosQcow2Image=$3
```
Comment (njagadish) [inline lib/nfp]: Is standalone configurator image support removed completely?

File: lib/nfp [hunk 4]
```diff
@@ -145,66 +146,156 @@ function create_nfp_gbp_resources {
...
-    HaproxyImageName=haproxy
-    if [ ! -z "$1" -a "$1" != " " ]; then
-        echo "Uploading Image : $ConfiguratorImageName"
-        glance image-create --name $ConfiguratorImageName --disk-format qcow2  --container-format bare  --visibility public --file $ConfiguratorQcow2ImageName
-        echo "Uploading Image : $VyosImageName"
-        glance image-create --name $VyosImageName --disk-format qcow2  --container-format bare  --visibility public --file $VyosQcow2Image
-        echo "Uploading Image : $HaproxyImageName"
-        glance image-create --name $HaproxyImageName --disk-format qcow2  --container-format bare  --visibility public --file $HaproxyQcow2Image
+    [[ $5 = "" ]] && VISIBILITY_FLAG=False
+    if [[ $VISIBILITY_FLAG = False ]]; then
```
Comment (njagadish) [inline lib/nfp:192]: We need to support launching two different VMs for configurator and visibility later.

File: lib/nfp [hunk 5]
```diff
@@ -29,7 +29,7 @@ function namespace_delete {
 function namespace_create {
 
     source $1/openrc neutron service
-    SERVICE_MGMT_NET="l2p_svc_management_ptg"
+    SERVICE_MGMT_NET="l2p_service_management_l2p"
```
Comment (rajendramachani) [inline lib/nfp:32]: Why this name is changed ?
The old name is meaningful. The new one has duplicate substring l2p

**Comment by rajendramachani** [comment]
Deprecated changes

**Pull request closed without merging**